scripts/pre_commit_linter.py

```diff
@@ -732,7 +732,7 @@ def _pre_commit_linter(all_files):
     for process in linting_processes:
         # Require timeout parameter to prevent against endless waiting for the
         # linting function to return.
-        process.join(timeout=200)
+        process.join(timeout=600)
```

Comment:
Hmm, maybe we should make this value change depending on the number of files to lint?

Reply:
I agree. We earlier used to have 600 as the timeout value therefore, I went with this. How about completely removing this parameter -- that is, removing join altogether? Even if we go on endless wait by chance, the build will ultimately time-out because of Travis's timeout.

Reply:
I guess it depends on the overall timeout duration, I don't want to see developers waiting the maximum time either :/

Let's start with an experimental fix which uses the number of files, and if that ends up not working we'll use travis's timeout.

Reply:
Sounds good! Let me make the necessary changes then. Thanks!

Reply:
@brianrodri: Atm, we have 1640 files in all to lint. The timeout of 200 seems to be insufficient. Do you have some proportion of files to be set as timeout in mind? Thanks!